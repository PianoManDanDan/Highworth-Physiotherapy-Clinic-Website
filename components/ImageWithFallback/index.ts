export * from './ImageWithFallback';
export * from './ImageWithFallbackProps';
