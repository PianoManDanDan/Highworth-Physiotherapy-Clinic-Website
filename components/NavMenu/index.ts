export * from './NavMenu';
export * from './subcomponents';