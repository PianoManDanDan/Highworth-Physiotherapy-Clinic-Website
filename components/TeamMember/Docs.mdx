import { Meta, Title, Primary, Controls, Canvas } from '@storybook/blocks';
import * as TeamMemberStories from './TeamMember.stories';

<Meta of={TeamMemberStories} />

<Title />

Provides a brief biography for staff. Physiotherapists have extra information
with their biography (qualifications, specialisms, etc.) so they have a slightly
different layout and appearance to their `TeamMember` component.

<Primary />

## Props

<Controls />

## Stories

### Physiotherapist

Physiotherapist includes information on the individual's qualifications,
specialisms and graduation dates, as well as biographical information
about them.

<Canvas of={TeamMemberStories.physiotherapist} />

### Front desk staff

Front desk staff only have biographical information.

<Canvas of={TeamMemberStories.frontDeskStaff} />
