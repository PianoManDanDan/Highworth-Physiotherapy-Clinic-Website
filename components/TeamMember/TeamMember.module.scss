@use '../../styles/mixins/media-query' as *;

.team-member {
  display: flex;
  flex-direction: column;

  @include media-query(small) {
    flex-direction: row;
  };

  &__picture {
    align-self: center;
    margin: 0 1rem;

    @include media-query(small) {
      align-self: flex-start;
      margin-top: 1.25em;
    };
  };
};
