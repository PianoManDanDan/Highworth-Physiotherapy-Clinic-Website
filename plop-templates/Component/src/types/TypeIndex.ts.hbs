export * from './{{pascalCase name}}Types';