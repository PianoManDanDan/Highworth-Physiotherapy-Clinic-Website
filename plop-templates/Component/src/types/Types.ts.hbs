export type {{pascalCase name}}Props = {
	property: string;
};