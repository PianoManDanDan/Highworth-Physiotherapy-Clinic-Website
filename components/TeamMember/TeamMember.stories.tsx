import { StoryObj, Meta } from '@storybook/react';
import { TeamMember } from 'components/TeamMember';
import { Person } from 'types/Person';

export default {
  component: TeamMember,
} as Meta;

const sharedProps: Partial<Person> = {
  name: 'John Smith',
  image: {
    src: 'https://picsum.photos/200',
    alt: 'team member',
  },
  bio: [
    'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Maecenas non sodales nisi. Nulla non ex at libero accumsan porttitor. Donec ultrices ullamcorper tortor ac dignissim. Donec rhoncus turpis nec lorem congue finibus. Donec euismod, diam ut sagittis euismod, magna justo mollis tortor, in mattis risus neque sit amet arcu. Mauris ut massa viverra est semper ullamcorper sit amet nec quam.',
    'Etiam aliquet et diam vitae cursus. Donec consectetur tristique lobortis. In vel rhoncus metus. Pellentesque et condimentum nulla. Quisque ut iaculis mauris. Nam id tellus mi. Aliquam placerat mauris eget risus malesuada ultricies. Ut eget mauris at nisl vulputate fermentum.',
    'Fusce dignissim viverra tellus, non suscipit tortor egestas sed. Praesent finibus gravida arcu. Nunc risus ligula, cursus vitae tellus quis, venenatis convallis turpis. Cras dignissim eleifend fermentum. Quisque aliquet lobortis efficitur. In lobortis ultricies odio quis auctor. Nunc iaculis nisi tempor malesuada vulputate. Integer rhoncus risus a velit interdum lacinia.',
    'Aenean at ullamcorper dui. Donec non dolor mauris. Duis ultrices lorem et nibh molestie, et facilisis lorem placerat. Sed sollicitudin, felis quis pulvinar eleifend, enim leo fermentum dolor, sed mattis nunc nisl vel urna. Curabitur arcu nibh, rhoncus quis suscipit vitae, faucibus volutpat mi. Nulla commodo est id risus auctor laoreet. Etiam id augue magna. Nulla nibh neque, lobortis quis nibh nec, commodo hendrerit lectus.'
  ],
}

export const physiotherapist: StoryObj<typeof TeamMember> = {
  args: {
    ...sharedProps,
    qualification: 'MCSP',
    yearQualified: 2005,
    specialities: [
      'Head',
      'Neck',
      'Spine',
      'Knee',
    ],
  }
};

export const frontDeskStaff: StoryObj<typeof TeamMember> = {
  args: sharedProps as Person,
  parameters: {
    controls: {
      exclude: ['qualification', 'yearQualified', 'specialities']
    },
  },
};
