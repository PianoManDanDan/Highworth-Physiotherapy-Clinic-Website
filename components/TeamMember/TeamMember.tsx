import { ImageWithFallback } from 'components/ImageWithFallback';
import { Person } from 'types';
import styles from './TeamMember.module.scss';
import teamMemberFallback from 'public/assets/people/placeholder.png';

export const TeamMember: React.FC<Person> = ({
  name,
  image,
  bio,
  qualification,
  yearQualified,
  specialities,
}) => {
  image.height = 160;
  image.width = 160;

  return (
    <div className={styles['team-member']}>
      <div className={styles['team-member__picture']}>
        <ImageWithFallback
          src={image.src}
          alt={image.alt}
          height={image.width}
          width={image.height}
          fallbackSrc={teamMemberFallback}
        />
      </div>
      <div>
        <h2>{name} {qualification || null}</h2>
        {
          yearQualified
          ? <p>Qualified {yearQualified}</p>
          : null
        }
        {
          specialities && specialities.length !== 0
          ? <p>Speciality: {specialities.join(', ')}</p>
          : null
        }
        {bio.map((paragraph, index) => <p key={index}>{paragraph}</p>)}
      </div>
    </div>
  );
};

export default TeamMember;
