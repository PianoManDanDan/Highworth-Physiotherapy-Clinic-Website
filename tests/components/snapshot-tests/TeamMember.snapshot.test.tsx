import React from 'react';
import { render } from '@testing-library/react';
import { TeamMember } from '../../../components/TeamMember';
import { Person } from '../../../types';

describe('TeamMember Component snapshot test', () => {
  describe('When passed only required props', () => {
    test('it renders correctly, with the Person\'s name, image and bio', () => {
      // Arrange
      const teamMemberData: Person = {
        name: 'Team member name test',
        image: {
          src: '/Team member image src test',
          altText: 'Team member image alt test',
        },
        bio: ['Team member bio paragraph 1 test', 'Team member bio paragraph 2 test'],
      };
      
      const { asFragment } = render(
        <TeamMember {...teamMemberData} />
      );
  
      // Act
      const renderedTeamMember = asFragment();
  
      // Assert
      expect(renderedTeamMember).toMatchSnapshot();
    });
  });
  describe('When passed all props', () => {
    test('it renders correctly, with the Person\'s name, image, qualification, year of qualification, specialities and bio', () => {
      // Arrange
      const teamMemberData: Person = {
        name: 'Team member name test',
        image: {
          src: '/Team member image src test',
          altText: 'Team member image alt test',
        },
        qualification: 'Team member qualification test',
        yearQualified: 1234,
        specialities: ['Team member specialities 1 test', 'Team member specialities 2 test'],
        bio: ['Team member bio paragraph 1 test', 'Team member bio paragraph 2 test'],
      };

      const { asFragment } = render(
        <TeamMember {...teamMemberData} />
      );

      // Act
      const renderedTeamMember = asFragment();

      // Assert
      expect(renderedTeamMember).toMatchSnapshot();
    });
  });
});
