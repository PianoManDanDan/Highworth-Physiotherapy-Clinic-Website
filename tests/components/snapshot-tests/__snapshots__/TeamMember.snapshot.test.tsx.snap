// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TeamMember Component snapshot test When passed all props it renders correctly, with the Person's name, image, qualification, year of qualification, specialities and bio 1`] = `
<DocumentFragment>
  <div>
    <div>
      <div
        style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
      >
        <div
          style="box-sizing: border-box; display: block; max-width: 100%;"
        >
          <img
            alt=""
            aria-hidden="true"
            role="presentation"
            src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMTYwIiBoZWlnaHQ9IjE2MCIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB2ZXJzaW9uPSIxLjEiLz4="
            style="max-width: 100%; display: block; margin: 0px; padding: 0px;"
          />
        </div>
        <noscript />
        <img
          alt="Team member image alt test"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
        />
      </div>
    </div>
    <div>
      <h2>
        Team member name test Team member qualification test
      </h2>
      <p>
        Qualified 1234
      </p>
      <p>
        Speciality: Team member specialities 1 test, Team member specialities 2 test
      </p>
      <p>
        Team member bio paragraph 1 test
      </p>
      <p>
        Team member bio paragraph 2 test
      </p>
    </div>
  </div>
</DocumentFragment>
`;

exports[`TeamMember Component snapshot test When passed only required props it renders correctly, with the Person's name, image and bio 1`] = `
<DocumentFragment>
  <div>
    <div>
      <div
        style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
      >
        <div
          style="box-sizing: border-box; display: block; max-width: 100%;"
        >
          <img
            alt=""
            aria-hidden="true"
            role="presentation"
            src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMTYwIiBoZWlnaHQ9IjE2MCIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB2ZXJzaW9uPSIxLjEiLz4="
            style="max-width: 100%; display: block; margin: 0px; padding: 0px;"
          />
        </div>
        <noscript />
        <img
          alt="Team member image alt test"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
        />
      </div>
    </div>
    <div>
      <h2>
        Team member name test 
      </h2>
      <p>
        Team member bio paragraph 1 test
      </p>
      <p>
        Team member bio paragraph 2 test
      </p>
    </div>
  </div>
</DocumentFragment>
`;
